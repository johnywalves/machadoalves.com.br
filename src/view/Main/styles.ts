import styled from 'styled-components'

export const Forehead = styled.div`
  width: 100%;
  background-color: var(--forehead-bg-color);
  padding: 6rem 0;
`

export const Content = styled.div`
  width: 80rem;
  margin: 0 auto;
  display: flex;
  flex-direction: row;
  justify-content: space-between;

  & div {
    width: 46rem;
    text-align: right;
    display: flex;
    flex-direction: column;
    align-items: flex-end;
    justify-content: center;

    & h2 {
      color: var(--highlight-text);
      font-size: 6.4rem;
      margin-bottom: 2rem;
    }

    & p {
      font-size: 2.4rem;
      font-weight: 300;
    }
  }
`
