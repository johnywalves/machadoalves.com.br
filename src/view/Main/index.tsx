import WrapperPage from 'components/WrapperPage'

import Reading from './reading'
import * as S from './styles'

const Main = () => (
  <WrapperPage>
    <S.Forehead>
      <S.Content>
        <Reading />
        <div>
          <h2>Lorem ipsum dolor sit amet</h2>
          <p>
            Nam vitae luctus felis, nec commodo massa. Fusce dapibus dolor vel
            mi maximus, elementum sollicitudin odio
          </p>
        </div>
      </S.Content>
    </S.Forehead>
  </WrapperPage>
)

export default Main
