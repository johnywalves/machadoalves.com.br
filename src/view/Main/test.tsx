import { render, screen } from '@testing-library/react'

import Main from '.'

describe('<Main />', () => {
  it('should render the heading', () => {
    const { container } = render(<Main />)

    expect(
      screen.getByRole('heading', { name: /machado alves/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the colors correctly', () => {
    const { container } = render(<Main />)

    expect(container.firstChild?.firstChild).toHaveStyleRule(
      'background-color',
      'var(--icon-bg-color)'
    )
  })
})
