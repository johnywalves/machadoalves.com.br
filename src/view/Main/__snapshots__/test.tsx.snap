// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
.c0 {
  height: 8rem;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c1 {
  height: 5.2rem;
  width: 5.2rem;
  margin: 0 1.2rem;
  border-radius: 100%;
  background-color: var(--icon-bg-color);
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c2 h1 {
  font-size: 3.6rem;
}

.c2 small {
  font-size: 1.2rem;
}

<header
  class="c0"
>
  <div
    class="c1"
  />
  <div
    class="c2"
  >
    <h1>
      Machado Alves
    </h1>
    <small>
      Históricas curtas escritas por mim, amigos e domínio público
    </small>
  </div>
</header>
`;
